azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/extensions/AzureComputeImageExtension.java

```diff
@@ -0,0 +1,149 @@
+      api.getVirtualMachineApi(group).stop(id);
+      //Poll until resource is ready to be used
+      boolean jobDone = Predicates2.retry(new Predicate<String>() {
+         @Override
+         public boolean apply(String name) {
+            String status = "";
+            List<VirtualMachineInstance.VirtualMachineStatus> statuses = api.getVirtualMachineApi(group).getInstanceDetails(name).statuses();
+            for (int c = 0; c < statuses.size(); c++) {
+               if (statuses.get(c).code().substring(0, 10).equals("PowerState")) {
+                  status = statuses.get(c).displayStatus();
+                  break;
+               }
+            }
+            if (status.equals("VM stopped")) {
+               return true;
+            }
+            return false;
+         }
+      }, 60 * 4 * 1000).apply(id);
+
+      return userExecutor.submit(new Callable<Image>() {
+         @Override
+         public Image call() throws Exception {
+            api.getVirtualMachineApi(group).generalize(id);
+
+            final String[] disks = new String[2];
+            URI uri = api.getVirtualMachineApi(group).capture(id, cloneTemplate.getName(), CONTAINER_NAME);
+            if (uri != null) {
+               if (imageAvailablePredicate.apply(uri)) {
+                  try {
+                     List<ResourceDefinition> definitions = api.getJobApi().captureStatus(uri);
+                     if (definitions != null) {
+                        for (ResourceDefinition definition : definitions) {
+                           LinkedTreeMap<String, String> properties = (LinkedTreeMap<String, String>) definition.properties();
+                           Object storageObject = properties.get("storageProfile");
+                           LinkedTreeMap<String, String> properties2 = (LinkedTreeMap<String, String>) storageObject;
+                           Object osDiskObject = properties2.get("osDisk");
+                           LinkedTreeMap<String, String> osProperties = (LinkedTreeMap<String, String>) osDiskObject;
+                           Object dataDisksObject = properties2.get("dataDisks");
+                           ArrayList<Object> dataProperties = (ArrayList<Object>) dataDisksObject;
+                           LinkedTreeMap<String, String> datadiskObject = (LinkedTreeMap<String, String>) dataProperties.get(0);
+
+                           disks[0] = osProperties.get("name");
+                           disks[1] = datadiskObject.get("name");
+
+                           VirtualMachine vm = api.getVirtualMachineApi(group).get(id);
+                           String location = vm.location();
+                           final VMImage ref = VMImage.create(CUSTOM_IMAGE_PREFIX + group, CUSTOM_IMAGE_PREFIX + storageAccountName, disks[0], disks[1], location, false);
+                           return imageReferenceToImage.apply(ref);
+                        }
+                     }
+                  } catch (Exception e) {
```

Comment:
What can fail here? Is there something we would want to log to give users proper feedback of what is going on?